### DISCUSSION TITLE
Problemes with PWM and duty response

### DISCUSSION THREAD

**Discussion by Baluar81** [arendst/Tasmota#14371] [Q&A] [answered]
Hi there!

I'm using an ESP32 and SSR to manage photovoltaic's excedents with PWM (#13123). I have realised there is an error, or a non corret work, when change the PWM's duty the frequency of the PWM. While in the terminal I can see there is a change, it doenst work trough GPIO. The matter is if I change de PWM's frequency, it seems keeping the duty when in fact it doesnt change the GPIO's output.

I attach an sceenshot and a table to explain better this error.
![photo_2022-01-05_17-26-09](https://user-images.githubusercontent.com/75850616/148809195-94afadbe-4002-41a1-bcc7-c5277daa1f64.jpg)

**Comment by s-hadinger** [comment, marked as answer]
Tasmota is not designed to change pwm frequency on the fly. A reboot is expected when you change the frequency. Otherwise you may see weird behavior
